bgpd/bgp_zebra.c

```diff
@@ -394,12 +396,23 @@ static int bgp_interface_address_delete(int command, struct zclient *zclient,
 			   ifc->ifp->name, buf);
 	}
 
+	/* Update the router id if the interface address being deleted is
+	 * same as the bgp router id
```

Comment:
This shouldn't be necessary. If an interface address is deleted and this address is the current router ID chosen by zebra, then zebra will update its router ID and notify the client daemons. There's no need to replicate this logic here. Please check the `router_id_del_address()` function in zebra.